src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java

```diff
@@ -0,0 +1,157 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.services;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrData;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrDataField;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OcrFieldRetrieverTest {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    public void setUp() {
+        mapper.registerModule(new JavaTimeModule());
+    }
+
+    @Test
+    public void should_return_error_when_field_required_but_not_found() {
+        OcrData ocrData = ocrData(new OcrDataField("field_1", "value"));
+
+        ResultOrErrors<String> result = ocrFieldRetriever(ocrData).getFieldValue(
+            "non-existing-field",
+            true,
+            String.class
```

Comment:
Would be nice to show this for Boolean and Date class too. I am not too strong on this but will leave it upto you :)

Reply:
I'll cover primitive types

Reply:
Done for Integer, Boolean, LocalDate and String.